File: media_turma.py
# Leitura de duas notas de prova para 5 alunos e cálculo da média da turma em cada prova

NUM_ALUNOS = 5

notas_prova1 = []
notas_prova2 = []

print("Informe as notas de 5 alunos (0-10).\n")

for i in range(1, NUM_ALUNOS + 1):
    while True:
        try:
            n1 = float(input(f"Aluno {i} - nota da prova 1: "))
            if not (0 <= n1 <= 10):
                raise ValueError
            break
        except ValueError:
            print("Valor inválido. Digite um número entre 0 e 10.")

    while True:
        try:
            n2 = float(input(f"Aluno {i} - nota da prova 2: "))
            if not (0 <= n2 <= 10):
                raise ValueError
            break
        except ValueError:
            print("Valor inválido. Digite um número entre 0 e 10.")

    notas_prova1.append(n1)
    notas_prova2.append(n2)

media_prova1 = sum(notas_prova1) / NUM_ALUNOS
media_prova2 = sum(notas_prova2) / NUM_ALUNOS

print("\nMédia da turma:")
print(f" - Prova 1: {media_prova1:.2f}")
print(f" - Prova 2: {media_prova2:.2f}")

File: media_turma.c
#include <stdio.h>

int main(void) {
    const int NUM_ALUNOS = 5;
    float nota1, nota2;
    float soma1 = 0.0f, soma2 = 0.0f;

    printf("Informe as notas de 5 alunos (0-10)\n\n");

    for (int i = 1; i <= NUM_ALUNOS; i++) {
        do {
            printf("Aluno %d - nota da prova 1: ", i);
            if (scanf("%f", &nota1) != 1) {
                nota1 = -1.0f;
                // limpar entrada inválida
                int c;
                while ((c = getchar()) != '\n' && c != EOF) {}
            }
        } while (nota1 < 0.0f || nota1 > 10.0f);

        do {
            printf("Aluno %d - nota da prova 2: ", i);
            if (scanf("%f", &nota2) != 1) {
                nota2 = -1.0f;
                int c;
                while ((c = getchar()) != '\n' && c != EOF) {}
            }
        } while (nota2 < 0.0f || nota2 > 10.0f);

        soma1 += nota1;
        soma2 += nota2;
    }

    printf("\nMédia da turma:\n");
    printf(" - Prova 1: %.2f\n", soma1 / NUM_ALUNOS);
    printf(" - Prova 2: %.2f\n", soma2 / NUM_ALUNOS);

    return 0;
}
